zingolib/src/wallet/traits.rs

```diff
@@ -450,7 +449,7 @@ pub trait ReceivedNoteAndMetadata: Sized {
     fn witnessed_position_mut(&mut self) -> &mut Option<Position>;
 }
 
-impl ReceivedNoteAndMetadata for ReceivedSaplingNoteAndMetadata {
+impl NoteInterface for SaplingNote {
```

Comment:
although a sapling note is probably technically `zcash_primitives::sapling::Note` i do agree with this name. in terms of our wallet i think it makes sense that the metadata is part of our note struct